module_utils/foreman_helper.py

```diff
@@ -51,22 +53,189 @@ def filter_module_params(self):
     def get_server_params(self):
         return (self._foremanapi_server_url, self._foremanapi_username, self._foremanapi_password, self._foremanapi_verify_ssl)
 
+
+class ForemanAnsibleModule(ForemanBaseAnsibleModule):
+
+    def check_requirements(self):
+        if not HAS_NAILGUN:
+            self.fail_json(msg='The nailgun Python module is required',
+                           exception=NAILGUN_IMP_ERR)
+
     def connect(self, ping=True):
         entity_mixins.DEFAULT_SERVER_CONFIG = ServerConfig(
             url=self._foremanapi_server_url,
             auth=(self._foremanapi_username, self._foremanapi_password),
             verify=self._foremanapi_verify_ssl,
         )
 
+    def ping(self):
+        try:
+            return Ping().search_json()
+        except Exception as e:
+            self.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+
+
+class ForemanApypieAnsibleModule(ForemanBaseAnsibleModule):
+
+    def check_requirements(self):
+        if not HAS_APYPIE:
+            self.fail_json(msg='The apypie Python module is required',
+                           exception=APYPIE_IMP_ERR)
+
+    def connect(self, ping=True):
+        try:
+            self.foremanapi = apypie.Api(uri=self._foremanapi_server_url,
+                                         username=self._foremanapi_username,
+                                         password=self._foremanapi_password,
+                                         api_version=2,
+                                         verify_ssl=self._foremanapi_verify_ssl)
+        except Exception as e:
+            self.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+
         if ping:
             self.ping()
 
     def ping(self):
         try:
-            return Ping().search_json()
+            return self.foremanapi.resource('home').call('status')
```

Comment:
This is a change from `/katello/api/ping` to `api/status`. Which I'm fine with, but we should be aware..

We should probably do some logic based on if it's a foreman_* or katello_* module, but we could do that in another PR later

Reply:
Yeah, we can have a `KatelloApypieAnsibleModule` that overrides this :)